diff --git a/scripts/build.py b/scripts/build.py
--- a/scripts/build.py
+++ b/scripts/build.py
@@ -28,36 +28,11 @@
     raise Exception(f"Unsupported arch: {uname}")
 
 
 arch = detect_arch()
-
-def download_so():
-    if (Path(arch["libdir"]) / arch["so_name"]).exists():
-        print(".so files alreay downloaded.")
-        return
-
-    file = "libcurl-impersonate.tar.gz"
-    url = (
-        f"https://github.com/yifeikong/curl-impersonate/releases/download/"
-        f"v{__version__}/libcurl-impersonate-v{__version__}"
-        f".{arch['so_arch']}-{arch['sysname']}.tar.gz"
-    )
-
-    print(f"Downloading libcurl-impersonate-chrome from {url}...")
-    urlretrieve(url, file)
-
-    print("Unpacking downloaded files...")
-    os.makedirs(arch["libdir"], exist_ok=True)
-    shutil.unpack_archive(file, arch["libdir"])
-
-    if arch["system"] == "Windows":
-        shutil.copy2(f"{arch['libdir']}/libcurl.dll", "curl_cffi")
-
-
 ffibuilder = FFI()
 system = platform.system()
 root_dir = Path(__file__).parent.parent
-download_so()
 
 
 ffibuilder.set_source(
     "curl_cffi._wrapper",

