diff --git a/src/Network/HTTP2/Client/Channels.hs b/src/Network/HTTP2/Client/Channels.hs
index 2644b45..f2c04f0 100644
--- a/src/Network/HTTP2/Client/Channels.hs
+++ b/src/Network/HTTP2/Client/Channels.hs
@@ -11,12 +11,14 @@ module Network.HTTP2.Client.Channels (
 
 import           Control.Concurrent.Chan.Lifted (Chan, readChan, newChan, writeChan)
 import           Control.Exception.Lifted (Exception, throwIO)
-import           Network.HTTP2.Frame (StreamId, FrameHeader, FramePayload, FrameType, framePayloadToFrameType, streamId)
+import           Network.HTTP2.Frame (StreamId, FrameHeader, FramePayload, FrameType, FrameDecodeError, framePayloadToFrameType, streamId)
 
 import           Network.HTTP2.Client.Exceptions
 
 type FramesChan e = Chan (FrameHeader, Either e FramePayload)
 
+instance Exception FrameDecodeError
+
 whenFrame
   :: Exception e
   => (FrameHeader -> FramePayload -> Bool)
