From 20eb0b279e1b9501b593aeeb8ff321313f21939c Mon Sep 17 00:00:00 2001
From: Alexey Chernov <4ernov@gmail.com>
Date: Fri, 22 May 2020 00:59:01 +0300
Subject: [PATCH 2/2] Use platform-dependent `LIBDIR` in `TorchConfig.cmake.in`

---
 cmake/TorchConfig.cmake.in | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/cmake/TorchConfig.cmake.in b/cmake/TorchConfig.cmake.in
index ee302e0e16..47807f982e 100644
--- a/cmake/TorchConfig.cmake.in
+++ b/cmake/TorchConfig.cmake.in
@@ -46,45 +46,45 @@ else()
   set(TORCH_LIBRARIES torch)
 endif()
 
-find_library(C10_LIBRARY c10 PATHS "${TORCH_INSTALL_PREFIX}/lib")
+find_library(C10_LIBRARY c10 PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
 list(APPEND TORCH_LIBRARIES ${C10_LIBRARY})
 
 # We need manually add dependent libraries when they are not linked into the
 # shared library.
 # TODO: this list might be incomplete.
 if(NOT @BUILD_SHARED_LIBS@)
-  find_library(TORCH_CPU_LIBRARY torch_cpu PATHS "${TORCH_INSTALL_PREFIX}/lib")
+  find_library(TORCH_CPU_LIBRARY torch_cpu PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
   list(APPEND TORCH_LIBRARIES ${TORCH_CPU_LIBRARY})
 
   if(@USE_NNPACK@)
-    find_library(NNPACK_LIBRARY nnpack PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(NNPACK_LIBRARY nnpack PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
     list(APPEND TORCH_LIBRARIES ${NNPACK_LIBRARY})
   endif()
 
   if(@USE_PYTORCH_QNNPACK@)
-    find_library(PYTORCH_QNNPACK_LIBRARY pytorch_qnnpack PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(PYTORCH_QNNPACK_LIBRARY pytorch_qnnpack PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
     list(APPEND TORCH_LIBRARIES ${PYTORCH_QNNPACK_LIBRARY})
   endif()
 
   if(@USE_XNNPACK@)
-    find_library(XNNPACK_LIBRARY XNNPACK PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(XNNPACK_LIBRARY XNNPACK PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
     list(APPEND TORCH_LIBRARIES ${XNNPACK_LIBRARY})
   endif()
 
   if(NOT @USE_INTERNAL_PTHREADPOOL_IMPL@)
-    find_library(PTHREADPOOL_LIBRARY pthreadpool PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(PTHREADPOOL_LIBRARY pthreadpool PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
     list(APPEND TORCH_LIBRARIES ${PTHREADPOOL_LIBRARY})
   endif()
 
   if(@INTERN_USE_EIGEN_BLAS@)
-    find_library(EIGEN_BLAS_LIBRARY eigen_blas PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    find_library(EIGEN_BLAS_LIBRARY eigen_blas PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
     list(APPEND TORCH_LIBRARIES ${EIGEN_BLAS_LIBRARY})
   endif()
 
-  find_library(CPUINFO_LIBRARY cpuinfo PATHS "${TORCH_INSTALL_PREFIX}/lib")
+  find_library(CPUINFO_LIBRARY cpuinfo PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
   list(APPEND TORCH_LIBRARIES ${CPUINFO_LIBRARY})
 
-  find_library(CLOG_LIBRARY clog PATHS "${TORCH_INSTALL_PREFIX}/lib")
+  find_library(CLOG_LIBRARY clog PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
   list(APPEND TORCH_LIBRARIES ${CLOG_LIBRARY})
 endif()
 
@@ -102,14 +102,14 @@ if (@USE_CUDA@)
       ${CUDA_TOOLKIT_ROOT_DIR}/lib/libnvToolsExt.dylib
       ${CUDA_LIBRARIES})
   else()
-    find_library(LIBNVTOOLSEXT libnvToolsExt.so PATHS ${CUDA_TOOLKIT_ROOT_DIR}/lib64/)
+    find_library(LIBNVTOOLSEXT libnvToolsExt.so PATHS ${CUDA_TOOLKIT_ROOT_DIR}/@CMAKE_INSTALL_LIBDIR@/)
     set(TORCH_CUDA_LIBRARIES
       ${CUDA_CUDA_LIB}
       ${CUDA_NVRTC_LIB}
       ${LIBNVTOOLSEXT}
       ${CUDA_LIBRARIES})
   endif()
-  find_library(C10_CUDA_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib")
+  find_library(C10_CUDA_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
   list(APPEND TORCH_CUDA_LIBRARIES ${C10_CUDA_LIBRARY})
   list(APPEND TORCH_LIBRARIES ${TORCH_CUDA_LIBRARIES})
 endif()
@@ -119,7 +119,7 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
   set(TORCH_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=@GLIBCXX_USE_CXX11_ABI@")
 endif()
 
-find_library(TORCH_LIBRARY torch PATHS "${TORCH_INSTALL_PREFIX}/lib")
+find_library(TORCH_LIBRARY torch PATHS "${TORCH_INSTALL_PREFIX}/@CMAKE_INSTALL_LIBDIR@")
 set_target_properties(torch PROPERTIES
     IMPORTED_LOCATION "${TORCH_LIBRARY}"
     INTERFACE_INCLUDE_DIRECTORIES "${TORCH_INCLUDE_DIRS}"
-- 
2.26.2

